src/test/com/wolfssl/provider/jsse/test/WolfSSLEngineTest.java

```diff
@@ -2074,5 +2074,327 @@ public void testDTLSv13EngineResumeSession()
+            String protocol = enabledProtocols.get(i);
+
+            if (protocol.equals("TLS") || protocol.contains("DTLS")) {
+                continue;
+            }
+
+            final boolean[] serverReady = new boolean[] { false };
+            final int[] serverPort = new int[] { 0 };
+            final Exception[] serverException = new Exception[] { null };
+            final Exception[] clientException = new Exception[] { null };
+
+            Thread serverThread = new Thread() {
+                public void run() {
+                    try {
+                        doNonBlockingIO(protocol, serverPort, serverReady,
+                            false);
+                    } catch (Exception e) {
+                        serverReady[0] = true;
+                        serverException[0] = e;
+                    }
+                }
+            };
+
+            serverThread.start();
+            while (!serverReady[0]) {
+                Thread.sleep(50);
+            }
+            if (serverException[0] != null) {
+                throw serverException[0];
+            }
+
+            Thread clientThread = new Thread() {
+                public void run() {
+                    try {
+                        doNonBlockingIO(protocol, serverPort, serverReady,
+                            true);
+                    } catch (Exception e) {
+                        clientException[0] = e;
+                    }
+                }
+            };
+
+            clientThread.start();
+
+            /* 5 second timeout */
+            serverThread.join(5000);
+            clientThread.join(5000);
+
+            if (serverThread.isAlive() || clientThread.isAlive()) {
+                error("\t... failed");
+                fail("Test timed out for " + protocol);
+            }
+
+            if (serverException[0] != null) {
+                error("\t... failed");
+                throw serverException[0];
+            }
+            if (clientException[0] != null) {
+                error("\t... failed");
+                throw clientException[0];
+            }
+        }
+
+        pass("\t... passed");
+    }
+
+    private void doNonBlockingIO(String protocol, int[] serverPort,
+        boolean[] serverReady, boolean isClient) throws Exception {
+
+        java.nio.channels.ServerSocketChannel ssc = null;
+        java.nio.channels.SocketChannel sc = null;
+
+        try {
+            SSLContext ctx = tf.createSSLContext(protocol, engineProvider);
+            SSLEngine engine = ctx.createSSLEngine(
+                "wolfSSL test", 11111);
+            engine.setUseClientMode(isClient);
+            if (!isClient) {
+                engine.setNeedClientAuth(false);
+            }
+
+            if (!isClient) {
+                ssc = java.nio.channels.ServerSocketChannel.open();
+                ssc.socket().bind(new java.net.InetSocketAddress(
+                    java.net.InetAddress.getLocalHost(), 0));
```

Comment:
Can we `import java.net.InetAddress` and just call `InetAddress.getLocalHost()` here?  Same with InetSocketAddress, and SocketChannel/InetSocketAddress/InetAddress below too.  And anywhere else in these changes a similar pattern occurs.